db.go

```diff
@@ -572,7 +614,21 @@ func (db *DB) Insert(list ...interface{}) error {
 // small wrapper around sql.DB.Query that returns a *squalor.Rows
 // instead.
 func (db *DB) Query(query interface{}, args ...interface{}) (*Rows, error) {
-	serializer, err := db.getSerializer(query)
+	if !db.Deadline.IsZero() {
+		// Set an execution deadline for the query.
+		remainingMillis := int64(db.Deadline.Sub(time.Now()) / time.Millisecond)
+		if remainingMillis > 0 {
+			var err error
+			query, err = db.DeadlineQueryRewriter(db, query, remainingMillis)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, fmt.Errorf("Query deadline exceeded by %d ms", -remainingMillis)
```

Comment:
[`context.DeadlineExceeded`](https://golang.org/pkg/context/#pkg-variables) is a standard error value to signify that the deadline has passed.